### ISSUE TITLE
Add Field<Result<Record[N]>>.mapping(Function[N]<T[N], E>): Field<List<E>> kotlin extension functions

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12945]
Our kotlin friends would certainly appreciate a few new extension functions of the sort::

```kotlin
inline fun <T1, E> Field<org.jooq.Result<Record1<T1>>>.mapping(f: Function1<T1, E>): Field<List<E>> = convertFrom { it.map(Records.mapping(f)) }
inline fun <T1, T2, E> Field<org.jooq.Result<Record2<T1, T2>>>.mapping(f: Function2<T1, T2, E>): Field<List<E>> = convertFrom { it.map(Records.mapping(f)) }
...
```

This would allow for using a `mapping()` function on `Field<Result<Record[N]>>` types, similar to the existing convenience `mapping()` function on `Row[N]` types, which are available to all language users, including Java users.

This would allow for writing:

```kotlin
data class X(val i: Int?, val j: Int?)

// This
dsl.select(multiset(select(i, j)).mapping(::X))

// As a short form for this
dsl.select(multiset(select(i, j)).convertFrom { it.map(Records.mapping(::X)) }
```

A Scala implicit is probably possible as well, to be defined.

**Comment by lukaseder** [comment]
We already have such utilities for Scala, e.g.

```scala
  implicit def asMapperFromArgList2[T1, T2, E](f: (T1, T2) => E): RecordMapper[Record2[T1, T2], E] = new RecordMapper[Record2[T1, T2], E] {
    def map(record: Record2[T1, T2]) = f(record.value1, record.value2)
  }
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0

**Comment by lukaseder** [comment]
Possible alternative implementation directly in the Java API: https://github.com/jOOQ/jOOQ/issues/13539
